controllers/search_controller.go

```diff
@@ -240,6 +240,11 @@ func (r *SearchReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		log.Error(err, "Global Search setup failed")
 		return *result, err
 	}
+	result, err = r.addEnvToSearchAPI(ctx, instance)
```

Comment:
This only applies to Global Search, so maybe it's better to move into `r.reconcileGlobalSearch()`

Reply:
This has to happen on each managed hub too. So, I kept it in the normal controller flow. If it is moved to reconcileGlobalSearch, it will be executed only on the global hub cluster, right?

Reply:
Should I move it outside the annotation `if else` flow in reconcileGlobalSearch?

Reply:
Ohh, ok, I didn't realize this happens in each managed hub too.